src/Resources/js/modules/slot-allocation-update-handler.ts

```diff
@@ -0,0 +1,36 @@
+interface SlotAllocationUpdate {
+    date: Date;
+    slotAllocation: {
+        // key: slot, value: allocation count
+        [key: string]: number;
+    };
+}
+
+export class SlotAllocationUpdateHandler {
+    static handleUpdate(data: SlotAllocationUpdate): void {
+        let $slotSelector = $(`#day-${data.date}-slots`);
+        if (1 !== $slotSelector.length) {
+            return;
+        }
+
+        for (const [slot, count] of Object.entries(data.slotAllocation)) {
+            let $slotOption = $slotSelector.find(`option[value="${slot}"]`);
+            if (1 !== $slotOption.length) {
+                continue;
+            }
+
+            const slotLimit = $slotOption.data('limit');
+            if (slotLimit > 0) {
+                const slotTitle = $slotOption.data('title');
+                const slotText = `${slotTitle} (${count}/${slotLimit})`;
+                $slotOption.text(slotText);
```

Comment:
can be combined

$slotOption.text(slotText)
                .prop('disabled', slotLimit <= count);

Reply:
Fixed